web/src/components/TabNavigation.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div>
+    <div class="border-t-md d-flex bg-grey-darken-3 align-center" density="compact">
+      <v-tabs
+        v-model="appStore.currentFactoryTabIndex"
+        color="deep-orange"
+      >
+        <v-tab
+          v-for="(item, index) in appStore.factoryTabs"
+          :key="item.name"
+          class="text-none"
+          :text="item.name"
+          :value="index"
+        />
+
+      </v-tabs>
+      <v-btn
+        color="grey-darken-3"
+        icon="fas fa-plus"
+        size="x-small"
+        variant="flat"
+        @click="appStore.addTab()"
+      />
+    </div>
+
+    <div class="bg-grey-darken-2 py-1 px-2 d-flex align-center justify-space-between" dark density="compact">
+      <input
+        v-model.lazy="appStore.currentFactoryTab.name"
+        class="pa-1 rounded tab-name"
+        @keyup.enter="onEnterTabName"
+      >
+
+      <div class="d-flex align-center h-100 ga-2">
+        <ShareButton />
+        <v-btn
+          v-if="appStore.factoryTabs.length > 1"
+          color="red rounded"
+          icon="fas fa-trash"
+          size="small"
+          variant="flat"
+          @click="appStore.removeCurrentTab()"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+  import { useAppStore } from '@/stores/app-store'
+
+  const appStore = useAppStore()
+
+  const onEnterTabName = (event: KeyboardEvent) => {
```

Comment:
I deliberately used `v-model.lazy` so it only updates on `change` and not on `input`.
I did this so the localStorage does not get written every time the user presses a key, because that felt like a bad idea.

Reply:
Ah i see, I've never encountered any issues with localStorage not updating the state, every time a factory value gets change it's written to localStorage as there's a watcher on the factory data.